### PR TITLE
preload image for Game_Interpreter

### PR DESCRIPTION
**Pull request by krmbn0576**
Search all commands when the `Game_Interpreter` is set up.
Then, if there is a command to load the image, request the image to be used ahead of time.

### CODE REVIEW

File: js/rpg_objects/Game_Interpreter.js [hunk 1]
```diff
@@ -1806,10 +1840,12 @@ Game_Interpreter.requestImages = function(list){
...
-            // Show Battle Animation
-            case 337:
-                if(params[1]) ImageManager.requestAnimation(params[1]);
-                break;
+            // Enemy Transform
+            case 336:
+                var enemy = $dataEnemies[params[1]];
+                var name = enemy.battlerName;
+                var hue = enemy.battlerHue;
+                if (name) ImageManager.requestEnemy(name, hue);
```
Comment (ghost) [inline js/rpg_objects/Game_Interpreter.js]: missing break;

File: js/rpg_objects/Game_Interpreter.js [hunk 2]
```diff
@@ -1782,20 +1812,24 @@ Game_Interpreter.requestImages = function(list){
                 });
                 break;
 
-            // Chage Battle Back
+            // Change Battle Back
             case 283:
+                if ($gameParty.inBattle()) {
```
Comment (ghost) [inline js/rpg_objects/Game_Interpreter.js:1817]: Why do you check inBattle?
Original code doesn't have that.

File: js/rpg_objects/Game_Interpreter.js [hunk 3]
```diff
@@ -1782,20 +1812,24 @@ Game_Interpreter.requestImages = function(list){
...
+                    if (params[0]) ImageManager.requestBattleback1(params[0]);
+                    if (params[1]) ImageManager.requestBattleback2(params[1]);
+                }
                 break;
 
             // Change Parallax
             case 284:
+                if (!$gameParty.inBattle() && params[0]) ImageManager.requestParallax(params[0]);
```
Comment (ghost) [inline js/rpg_objects/Game_Interpreter.js]: same above.

File: js/rpg_objects/Game_Interpreter.js (hunk 2 above)
Comment (krmbn0576) [reply js/rpg_objects/Game_Interpreter.js:1817]: These are certainly special behaviors.
They're used entirely only in `Scene_Battle`. (as for parallax, only `Scene_Map`)
In addition, the background and parallax are very large in size as compared with the images such as character, so I decline the load.

Comment (ghost) [reply js/rpg_objects/Game_Interpreter.js:1817]: for early check?
OK, This method must be smart, LGTM for this line :sushi:

File: js/rpg_objects/Game_Interpreter.js (hunk 1 above)
Comment (ghost) [reply js/rpg_objects/Game_Interpreter.js]: oops something wrong :-< sorry.

**Review by ghost — APPROVED** [review]

**Comment by krmbn0576** [comment]
I moved filename empty check to `ImageManager.requestBitmap` for convenience ;-)

**Comment by ghost** [comment]
OK,LGTM 🤖

**Pull request merged**